internal/driver/driver.go

```diff
@@ -187,86 +184,29 @@ func (d *Driver) Initialize(lc logger.LoggingClient, asyncCh chan<- *dsModels.As
 	return nil
 }
 
-func (d *Driver) watchForConfigChanges() error {
-	sdkFlags := flags.New()
-	sdkFlags.Parse(os.Args[1:])
-	cpUrl, err := url.Parse(sdkFlags.ConfigProviderUrl())
-	if err != nil {
-		return err
+func (d *Driver) updateWritableConfig(rawWritableConfig interface{}) {
+	updated, ok := rawWritableConfig.(*CustomConfig)
+	if !ok {
+		d.lc.Error("unable to update writable custom config: Can not cast raw config to type 'CustomConfig'")
+		return
 	}
 
-	cpPort := 8500
-	port := cpUrl.Port()
-	if port != "" {
-		cpPort, err = strconv.Atoi(port)
-		if err != nil {
-			cpPort = 8500
-		}
-	}
+	d.configMu.Lock()
+	oldSubnets := d.config.AppCustom.DiscoverySubnets
+	oldScanPort := d.config.AppCustom.ScanPort
+	d.config.AppCustom = *updated
+	d.configMu.Unlock()
 
-	configClient, err := configuration.NewConfigurationClient(types.ServiceConfig{
-		Host:     cpUrl.Hostname(),
-		Port:     cpPort,
-		BasePath: baseConsulPath,
-		Type:     strings.Split(cpUrl.Scheme, ".")[0],
-	})
-	if err != nil {
-		return err
+	if updated.DiscoverySubnets != oldSubnets || updated.ScanPort != oldScanPort {
+		d.lc.Info("discover configuration has changed! Discovery will be triggered momentarily.")
+		d.debouncedDiscover()
 	}
-
-	go func() {
-		errorStream := make(chan error)
-		defer close(errorStream)
-
-		updateStream := make(chan interface{})
-		defer close(updateStream)
-
-		cfg := driverConfiguration{}
-		configClient.WatchForChanges(updateStream, errorStream, &cfg, "/Driver")
-		d.lc.Info("watching for configuration changes...")
-
-		for {
-			select {
-			case <-d.done:
-				return
-
-			case ex := <-errorStream:
-				d.lc.Error(ex.Error())
-
-			case raw, ok := <-updateStream:
-				if !ok {
-					return
-				}
-
-				d.lc.Info("driver configuration has been updated!")
-				d.lc.Debug(fmt.Sprintf("raw: %+v", raw))
-
-				newCfg, ok := raw.(*driverConfiguration)
-				if ok {
-					d.configMu.Lock()
-					oldSubnets := d.config.DiscoverySubnets
-					oldScanPort := d.config.ScanPort
-					d.config = newCfg
-					d.configMu.Unlock()
-
-					if !reflect.DeepEqual(newCfg.DiscoverySubnets, oldSubnets) ||
-						newCfg.ScanPort != oldScanPort {
-						d.lc.Info("discover configuration has changed!")
-						d.debouncedDiscover()
-					}
-				} else {
-					d.lc.Warn("unable to decode incoming configuration from registry")
-				}
-			}
-		}
-	}()
-	return nil
 }
 
 // HandleReadCommands triggers a protocol Read operation for the specified device.
 func (d *Driver) HandleReadCommands(devName string, p protocolMap, reqs []dsModels.CommandRequest) ([]*dsModels.CommandValue, error) {
 	d.lc.Debug(fmt.Sprintf("LLRP-Driver.HandleReadCommands: "+
-		"device: %s protocols: %v reqs: %+v", devName, p, reqs))
+		"device: %s protocols: %+v reqs: %+v", devName, p, reqs))
```

Comment:
`p` is a `map`, no `+` needed.
```suggestion
		"device: %s protocols: %v reqs: %+v", devName, p, reqs))
```

Reply:
In this case is should be `%s` rather than `%v`

Reply:
`p` is a `protocolMap` which is a `map[string]ProtocolProperties` so it should use `%v`.
`devName` is already `%s`.

Reply:
Yep, I missed the %s already there and focused on the v's. ;-)

Reply:
removed + for `p`